src/org/zaproxy/zap/extension/ascanrules/TestPathTraversal.java

```diff
@@ -433,9 +435,18 @@ public void scan(HttpMessage msg, String param, String value) {
             // this is nice because it means we do not have to guess any file names, and would only require one
             // request to find the vulnerability 
             // but it would be foiled by simple input validation on "..", for instance.
-        } catch (IOException e) {
-            log.warn("Error scanning parameters for Path Traversal: " + e.getMessage(), e);
-        }
+		} catch (SocketTimeoutException ste) {
+			log.info("A timeout occurred while checking [" + msg.getRequestHeader().getMethod() + "] ["
+					+ msg.getRequestHeader().getURI() + "], parameter [" + param + "] for Path Traversal. "
+					+ "The currently configured timeout is: "
+					+ Integer.toString(Model.getSingleton().getOptionsParam().getConnectionParam().getTimeoutInSecs()));
+			if (log.isDebugEnabled()) {
+				log.debug("Caught " + ste.getClass().getName() + " " + ste.getMessage());
+			}
+		} catch (IOException e) {
+			log.info("An error occurred while checking [" + msg.getRequestHeader().getMethod() + "] ["
+					+ msg.getRequestHeader().getURI() + "], parameter [" + param + "] for Path Traversal.");

```

Comment:
Should log the exception's message too (otherwise we don't know what caused it).